src/box/iproto.cc

```diff
@@ -1280,6 +1286,18 @@ iproto_flush(struct iproto_connection *con)
 			end = &obuf_end;
 		}
 	}
+	if (!con->can_write) {
+		/* Receiving end was closed. Discard the output. */
+		*end = *begin;
+		/*
+		 * We should activate `con->input` in case it was stopped,
+		 * because of readahead limit is reached.
+		 */
+		if (! ev_is_active(&con->input) &&
+		    rlist_empty(&con->in_stop_list))
+			ev_feed_event(con->loop, &con->input, EV_READ);
```

Comment:
Let's return 0 instead of 1 in this case. Then this will be done by the calling function (iproto_connection_on_output).

Reply:
Fixed